javascript/index.js

```diff
@@ -28,13 +28,119 @@
 
 
 // Iteration 1 - using callbacks
-// ...
+
+getInstruction("mashedPotatoes", 0, (step1) => {
+  document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
+  getInstruction("mashedPotatoes", 1, (step2) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
+    getInstruction("mashedPotatoes", 2, (step3) => {
+      document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
+      getInstruction("mashedPotatoes", 3, (step4) => {
+        document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
+        getInstruction("mashedPotatoes", 4, (step5) => {
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
+          document.querySelector("#mashedPotatoes").innerHTML += `<li> Mashed Potatoes are ready</li>`
+          document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
+        }, (error) => console.log(error));
+      }, (error) => console.log(error));
+    }, (error) => console.log(error));
+  }, (error) => console.log(error));
+}, (error) => console.log(error));
+
+
 
 // Iteration 2 - using promises
-// ...
+
+obtainInstruction('steak', 0)
+  .then(response=> {
+    document.querySelector("#steak").innerHTML += `<li>${response}</li>`;
+  }).then(function() {
+    return obtainInstruction('steak', 1)
+  })
```

Comment:
You can have more logic/tasks  in one promise, as long as they do not overlap.
DOM manipulation, in our case updating innerHTML is synchronous, and will be completed right away, so below syntax is equivalent:

```
obtainInstruction('steak', 0)
  .then(response=> {
    document.querySelector("#steak").innerHTML += `<li>${response}</li>`;
    return obtainInstruction('steak', 1)
  })
```